extensions/src/common/types/framework/blocks.ts

```diff
@@ -9,8 +9,23 @@ export type ButtonBlock = () => InternalButtonKey;
 
 export type BlockMetadata<
   Fn extends BlockOperation,
+  TFunctions extends Array<(...args: any[]) => any> = [],
   TParameters extends any[] = Parameters<Fn> extends [...infer R, BlockUtility] ? R : Parameters<Fn>
-> = Type<ReturnType<Fn>> & Text<TParameters> & Arguments<TParameters>;
+> = Type<ReturnType<Fn>> & Text<TParameters> & Arguments<TParameters> & {
+  optionalFunctions?: TFunctions;
+};
+
+export type ArgTransformer = (...args: any[]) => any[];
+
+export type BlockOptions = {
+  transform?: ArgTransformer;
+  type?: { [key: string]: string };
+  name?: { [key: string]: string };
+};
+
+export type Config = {
```

Comment:
Did array type not work here? i.e. `(ArgTransformer | BlockOptions;)[]`

Also, we'll eventually want to give this a more obvious name